packages/common/src/utils/regex/RegexUtils.tsx

```diff
@@ -28,4 +28,19 @@ export const RegexUtils = {
     const matcher = new RegExp(substring, 'i');
     return matcher.test(testString);
   },
+  // Case-insensitive version of 'startsWith'
+  matchString: (substring: string, testString: string) => {
+    const matcher = new RegExp(`^${substring}`, 'i');
+    return matcher.test(testString);
+  },
+  // matches test string against an object's name or code properties
+  matchNameOrCode: (
+    item: { name: string; code: string },
+    searchString: string
```

Comment:
Could take this a step further and allow matching against *any* object?

Function would check for matches against any of the properties of the passed-in object, and when you call it you just pass an object with only the fields you want matched.

Reply:
Also -- once you've made any changes you want to do here, I'll write some tests for the new functions.

Reply:
> Function would check for matches against any of the properties of the passed-in object, and when you call it you just pass an object with only the fields you want matched.

This works, I think ;) 
```ts
matchObjectStrings: (object: { [key: string]: string }, searchString: string) =>
    Object.keys(object).some(key =>
      RegexUtils.matchSubstring(object[key] as string, searchString)
    ),
```

Reply:
oh that's good! I see what you mean now 😁 

Was just doing it and realised that we shouldn't.. filtering all string props would include `id` and therefore return unexpected results. potentially other props could be added if the fragment is updated. Safer to be explicit about the props to filter.

Reply:
Oh, I was thinking that when you call the function, you just pass in an subset of the object containing the fields you want to check.

E.g. you have an item `item` with `{id: 123, name: "Drug A", price: 2.34, code: "1234", description: "Some drug"}`, then you'd just call it like: `matchObjectStrings({code: item.code, name: item.name}, <searchString>)` to match against just the "name" and "code" fields.

I feel like eventually we'll likely have a lot of different filters matching different things, so it'd be good to handle all possibilities without writing a new function for each. Can always wait to generalise until we need it though.

Reply:
ah, right. I thought it was pretty neat just checking all string props.

I had this
```
  // returns true if the search string is contained in any of the properties specified by the given keys
  matchObjectProperties: function <T>(
    substring: string,
    object: T,
    keys: Array<keyof T>
  ) {
    keys.some(key => RegexUtils.includes(substring, String(object[key])));
  },
```

so that I could
```
  options.filter(option =>
    RegexUtils.matchObjectProperties(state.inputValue, option, ['name', 'code'])
  );
```

which does the same thing.. I think I understand now 😄

Reply:
Yup, an array of field names would work well. Can it be optional though, so it'll match *all* keys if not specified. (parameter default would be `Object.keys(object)` -- is that possible, to use another parameter to generate the default value for another? 🤷‍♂️)